src/Microsoft.AspNetCore.Authentication.OpenIdConnect/OpenIdConnectHandler.cs

```diff
@@ -475,6 +475,12 @@ protected virtual Task<bool> HandleSignOutCallbackAsync()
                 authorizationResponse = messageReceivedContext.ProtocolMessage;
                 properties = messageReceivedContext.Properties;
 
+                // if any of the error fields are set, throw error null
+                if (!string.IsNullOrEmpty(authorizationResponse.Error))
```

Comment:
Fair. I was worried about the state being dropped for some error responses and thus hiding the error response with a correlation failure. Not sure how often that happens in practice.